File: baekjoon/Silver/[S1] 별 찍기 - 10.c
#include <stdio.h>

void check(int i, int j, int n);


int main()
{
	int n;

	scanf("%d", &n);
	for (int i = 1; i <= n; i++)
	{
		for (int j = 1; j <= n; j++)
			check(i, j, n);
		printf("\n");
	}
	return 0;
}


void check(int i, int j, int n)
{
	int a = 1, b = 3, c = 2;
	int imod, jmod;

	while (b <= n)
	{
		imod = i % b, jmod = j % b;
		if (a < imod && imod <= c && a < jmod && jmod <= c)
		{
			printf(" ");
			return;
		}
		a *= 3, b *= 3, c *= 3;
	}
	printf("*");
}

File: baekjoon/Silver/[S1] 별 찍기 - 10.py
def check(i, j, k):
    a, b, c = 1, 3, 2
    while b <= k:
        if a < i % b <= c and a < j % b <= c: print(' ', end=''); return
        a *= 3; b *= 3; c *= 3
    print('*', end='')

n = int(input())
for i in range(1, n + 1):
    for j in range(1, n + 1):
        check(i, j, n)
    print()
